rest/config.go

```diff
@@ -1890,6 +1893,12 @@ func StartServer(ctx context.Context, config *StartupConfig, sc *ServerContext)
 
 	go sc.PostStartup()
 
+	base.ConsolefCtx(ctx, base.LevelInfo, base.KeyAll, "Starting diagnostic server on %s", config.API.DiagnosticInterface)
+	go func() {
+		if err := sc.Serve(ctx, config, diagnosticServer, config.API.DiagnosticInterface, CreateDiagnosticHandler(sc)); err != nil {
+			base.ErrorfCtx(ctx, "Error serving the Diagnostic API: %v", err)
```

Comment:
To test this, let's add a new test to `bootstrap_test.go` that looks like:

```
	config := BootstrapStartupConfigForTest(t)
	sc, closeFn := StartServerWithConfig(t, &config)
	...
```

We can abstract `doAdminBootstrapRequest` to `doBootstrapRequest` that would let you make a request to any server.

Having a package level variable like `var allServers = []string{publicServer, adminServers, metricsServer, diagnosticServer}` would be good for this test and for `WaitForRESTAPIs`.